Docs - Add privacy controls (#10040)
PostHog/posthog.com: contents/docs/product-analytics/privacy.mdx

Reviewer: EU cloud mention?

---
title: Privacy controls
---

PostHog offers a range of controls to limit what data is captured by product analytics. They are listed below in order of least to most restrictive.

> ## Disable sensitive information with auto-capture

Reviewer: ```suggestion
## Disable sensitive information with autocapture
```

If you're using [autocapture](/docs/product-analytics/autocapture), PostHog automatically attempts to prevent sensitive data capture. We specifically only collect the `name`, `id`, and `class` attributes from input tags.

> If there are specific elements you don't want to capture, add the [ph-no-capture](/docs/product-analytics/autocapture#preventing-sensitive-data-capture) class name.

Reviewer: ```suggestion
If there are specific elements you don't want to capture, add the [`ph-no-capture`](/docs/product-analytics/autocapture#preventing-sensitive-data-capture) class name.
```

```html
<button class='ph-no-capture'>Sensitive information here</button>
```

> Additionally, you can also set [ignore lists](/docs/product-analytics/autocapture#reducing-events-with-an-allow-and-ignorelist) in your PostHog config to prevent specific elements from being captured.

Reviewer: ```suggestion
Additionally, you can set [allow or ignore lists](/docs/product-analytics/autocapture#reducing-events-with-an-allow-and-ignorelist) in your PostHog config to prevent specific elements from being captured.
```

> ## Use the property filter app

Reviewer: Maybe `sanitize_properties` on the client is better? Its a new-ish feature and we don't have much docs about it.

Author: will document both

> You can use the [property filter app](/docs/cdp/property-filter#who-created-this-transformation) to prevent PostHog from certain properties on events. For example, you can configure the app to remove all GeoIP data from events.

Reviewer: ```suggestion
You can use the [property filter app](/docs/cdp/property-filter) to prevent PostHog from certain properties on events. For example, you can configure the app to remove all GeoIP data from events.
```

> We've also put together a [tutorial](/tutorials/property-filter) to help you get started with the app.

Author: This tutorial is out of date. I'll update in the next PR

---

PostHog offers a range of controls to limit what data is captured by product analytics. They are listed below in order of least to most restrictive.

## EU-cloud hosting

PostHog offers hosting on EU cloud. To use this, sign up at [eu.posthog.com](https://eu.posthog.com).

If you've already created a US cloud instance and wish to migrate ticket, you must raise a [support ticket in-app](https://us.posthog.com/home#supportModal) with the **Data pipelines** topic for the PostHog team to do this migration for you. This option is **only available to customers on the team or enterprise plan** as it requires significant engineering time.  

## Disable sensitive information with autocapture

If you're using [autocapture](/docs/product-analytics/autocapture), PostHog automatically attempts to prevent sensitive data capture. We specifically only collect the `name`, `id`, and `class` attributes from input tags.

If there are specific elements you don't want to capture, add the [`ph-no-capture`](/docs/product-analytics/autocapture#preventing-sensitive-data-capture) class name.

```html
<button class='ph-no-capture'>Sensitive information here</button>
```

## Sanitize properties

> You can sanitize properties on the client side by setting the `sanitize_properties` [config](/docs/libraries/js#config) option. This is a function that enables to modify the properties before they are sent to PostHog. For example:

Author: ```suggestion
You can sanitize properties on the client side by setting the `sanitize_properties` [config](/docs/libraries/js#config) option. This is a function that enables you to modify the properties before they are sent to PostHog. For example:
```

```js-web
posthog.init('<ph_project_api_key>', {
    api_host: '<ph_client_api_host>',
    sanitize_properties: function(properties, event) {
        if (properties['$ip']) {
            properties['$ip'] = null;
        }
        return properties;
    }
});
```

> ##

Reviewer: Missing title or typo?

> ##

Author: ```suggestion
## Sanitize properties on the client
```

Additionally, you can set [allow or ignore lists](/docs/product-analytics/autocapture#reducing-events-with-an-allow-and-ignorelist) in your PostHog config to prevent specific elements from being captured.

## Use the property filter app

You can use the [property filter app](/docs/cdp/property-filter) to prevent PostHog from certain properties on events. For example, you can configure the app to remove all GeoIP data from events.

We've also put together a [tutorial](/tutorials/property-filter) to help you get started with the app.

## Cookieless tracking

It's possible to use PostHog without cookies. Instead, PostHog can use in-memory storage. For more details on how to do this, read our tutorial on [how to set up cookieless tracking](/tutorials/cookieless-tracking).

## Complete opt-out

You can completely opt-out users from data capture. To do this, there are two options:

> 1. Opt users out by default by setting `opt_out_capturing_by_default` to `true` in your [PostHog config](/docs/libraries/js#config).

Reviewer: Because it depends on the SDK

```suggestion
1. Opt users out by default in your PostHog initialization config.
```

<MultiLanguage>

```js-web
posthog.init('<ph_project_api_key>', {
    opt_out_capturing_by_default: true,
});
```

```ios_swift        
let config = PostHogConfig(apiKey: <ph_project_api_key>, host: <ph_client_api_host>)
config.optOut = true
PostHogSDK.shared.setup(config)
```

```android
val config = PostHogAndroidConfig(
    apiKey = <ph_project_api_key>,
    host = <ph_client_api_host> 
)
config.optOut = true
PostHogAndroid.setup(this, config)
```

```react-native
posthog.init('<ph_project_api_key>', {
    opt_out_capturing_by_default: true,
});
```

</MultiLanguage>

> 2. Opt users out on a per-person basis by calling `posthog.opt_out_capturing()`.

Reviewer: ```suggestion
2. Opt users out on a per-person basis.
```
